File: CUTEProject/bundles/ch.hsr.ifs.testframework/src/ch/hsr/ifs/testframework/model/TestElement.java
/*******************************************************************************
 * Copyright (c) 2007-2011, IFS Institute for Software, HSR Rapperswil,
 * Switzerland, http://ifs.hsr.ch
 *
 * Permission to use, copy, and/or distribute this software for any
 * purpose without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 ******************************************************************************/
package ch.hsr.ifs.testframework.model;

import java.util.ArrayList;
import java.util.List;


public abstract class TestElement {

   protected List<ITestElementListener> listeners = new ArrayList<>();
   private ITestComposite               parent;

   public abstract String getName();

   public abstract TestStatus getStatus();

   public void addTestElementListener(ITestElementListener lis) {
      if (!listeners.contains(lis)) {
         listeners.add(lis);
      }
   }

   public void removeTestElementListener(ITestElementListener lis) {
      listeners.remove(lis);
   }

   protected void notifyListeners(NotifyEvent event) {
      for (ITestElementListener lis : listeners) {
         lis.modelCanged(this, event);
      }
   }

   public ITestComposite getParent() {
      return parent;
   }

   public void setParent(ITestComposite parent) {
      this.parent = parent;
   }

   public String getRerunName() {
      StringBuilder result = new StringBuilder();
      if (getParent() != null && getParent().getRerunName().length() > 0) {
         result.append(getParent().getRerunName());
         result.append('#');
      }
      result.append(getName());
      return result.toString();
   }

}

File: CUTEProject/bundles/ch.hsr.ifs.testframework/src/ch/hsr/ifs/testframework/model/TestElement.kt
/*******************************************************************************
 * Copyright (c) 2007-2011, IFS Institute for Software, HSR Rapperswil,
 * Switzerland, http://ifs.hsr.ch
 *
 * Permission to use, copy, and/or distribute this software for any
 * purpose without fee is hereby granted, provided that the above
 * copyright notice and this permission notice appear in all copies.
 ******************************************************************************/
package ch.hsr.ifs.testframework.model;

import java.util.ArrayList;


abstract class TestElement {

   private val listeners = mutableListOf<ITestElementListener>()
   private var parent: ITestComposite? = null

   public abstract fun getName(): String;

   public abstract fun getStatus(): TestStatus;

   fun addTestElementListener(lis: ITestElementListener) {
      if (!listeners.contains(lis)) {
         listeners.add(lis);
      }
   }

   fun removeTestElementListener(lis: ITestElementListener) {
      listeners.remove(lis);
   }

   fun notifyListeners(event: NotifyEvent) {
	   listeners.forEach{ it.modelCanged(this, event) }
   }

   fun getParent() = parent

   fun setParent(parent: ITestComposite) {
      this.parent = parent;
   }

   fun getRerunName(): String {
      val result = StringBuilder();
      parent?.let{
         if(it.getRerunName().length > 0) {
            result.append(it.getRerunName()).append('#')
         }
      }
      result.append(getName());
      return result.toString();
   }

}
